plutus-scb/src/Plutus/SCB/App.hs

```diff
@@ -0,0 +1,164 @@
+                                             runStdoutLoggingT)
+import           Control.Monad.Reader       (MonadReader, ReaderT (ReaderT), asks, runReaderT)
+import           Data.Aeson                 (FromJSON, ToJSON, eitherDecode)
+import qualified Data.Aeson.Encode.Pretty   as JSON
+import qualified Data.ByteString.Lazy.Char8 as BSL8
+import qualified Data.Text                  as Text
+import           Database.Persist.Sqlite    (retryOnBusy, runSqlPool)
+import           Eventful                   (commandStoredAggregate, getLatestStreamProjection,
+                                             serializedEventStoreWriter, serializedGlobalEventStoreReader,
+                                             serializedVersionedEventStoreReader)
+import           Eventful.Store.Sql         (jsonStringSerializer, sqlEventStoreReader, sqlGlobalEventStoreReader)
+import           Eventful.Store.Sqlite      (initializeSqliteEventStore, sqliteEventStoreWriter)
+import           Network.HTTP.Client        (defaultManagerSettings, newManager)
+import           Plutus.SCB.Core            (Connection (Connection), ContractCommand (InitContract, UpdateContract),
+                                             MonadContract, MonadEventStore, addProcessBus, dbConnect, invokeContract,
+                                             refreshProjection, runCommand, toUUID)
+import           Plutus.SCB.Types           (Config (Config),
+                                             SCBError (ContractCommandError, NodeClientError, WalletClientError),
+                                             dbConfig, nodeServerConfig, walletServerConfig)
+import           Servant.Client             (ClientEnv, ClientM, ServantError, mkClientEnv, runClientM)
+import           System.Exit                (ExitCode (ExitFailure, ExitSuccess))
+import           System.Process             (readProcessWithExitCode)
+import           Wallet.API                 (NodeAPI, WalletAPI, WalletDiagnostics, logMsg, ownPubKey, sign, slot,
+                                             startWatching, submitTxn, updatePaymentWithChange, watchedAddresses)
+
+------------------------------------------------------------
+data Env =
+    Env
+        { dbConnection    :: Connection
+        , walletClientEnv :: ClientEnv
+        , nodeClientEnv   :: ClientEnv
+        }
+
+newtype App a =
+    App
+        { unApp :: ExceptT SCBError (ReaderT Env (LoggingT IO)) a
+        }
+    deriving newtype ( Functor
+                     , Applicative
+                     , Monad
+                     , MonadLogger
+                     , MonadIO
+                     , MonadReader Env
+                     , MonadError SCBError
+                     )
+
+instance NodeAPI App where
+    submitTxn = void . runNodeClientM . NodeClient.addTx
+    slot = runNodeClientM NodeClient.getCurrentSlot
+
+instance WalletAPI App where
```

Comment:
Not sure whether we really need this instance. I might change/delete it when I'm done with the mock wallet implementation but it can stay for now.